sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -1125,22 +1125,34 @@ def test_uint8_predict(Est):
     est.predict(X)
 
 
-# TODO: Remove in v1.2
 @pytest.mark.parametrize(
-    "old_loss, new_loss",
+    "old_loss, new_loss, Estimator",
     [
-        ("least_squares", "squared_error"),
-        ("least_absolute_deviation", "absolute_error"),
+        # TODO(1.2): Remove
+        ("least_squares", "squared_error", HistGradientBoostingRegressor),
+        ("least_absolute_deviation", "absolute_error", HistGradientBoostingRegressor),
+        # TODO(1.3): Remove
+        ("auto", "log_loss", HistGradientBoostingClassifier),
+        ("binary_crossentropy", "binary_log_loss", HistGradientBoostingClassifier),
+        (
+            "categorical_crossentropy",
+            "multiclass_log_loss",
+            HistGradientBoostingClassifier,
+        ),
     ],
 )
-def test_loss_deprecated(old_loss, new_loss):
-    X, y = make_regression(n_samples=50, random_state=0)
-    est1 = HistGradientBoostingRegressor(loss=old_loss, random_state=0)
+def test_loss_deprecated(old_loss, new_loss, Estimator):
+    X, y = X_classification[:10], y_classification[:10]
+
+    if old_loss == "categorical_crossentropy":
+        y[0] = 3  # make sure it is multiclass
```

Comment:
Mutating a shared global variable is likely to cause very confusing failures in other tests.

Instead let's do something like:

```suggestion
    if old_loss == "categorical_crossentropy":
        X, y = X_multi_classification[:10], y_multi_classification[:10]
        assert len(np.unique(y)) > 2
    else:
        X, y = X_classification[:10], y_classification[:10]
```

Reply:
Or alternatively make a copy of `y_multi_classification` before mutating it if there is no 3 classes in the first 10 elements (unlikely).

Reply:
Good catch!